lib/Dialect/TritonGPU/IR/Ops.cpp

```diff
@@ -662,41 +661,41 @@ LogicalResult AsyncCopyGlobalToLocalOp::verify() {
   return success();
 }
 
-LogicalResult MemDescSubviewOp::verify() {
+LogicalResult MemDescIndexOp::verify() {
   auto srcTy = getSrc().getType();
   auto dstTy = getType();
-
   if (srcTy.getElementType() != dstTy.getElementType()) {
     return emitError("result element type must match desc element type");
   }
-  if (getOffsets().size() != srcTy.getRank()) {
-    return emitError("offsets must have the same rank as input");
+  bool is1D =
+      srcTy.getRank() == 1 && dstTy.getRank() == 1 && dstTy.getDimSize(0) == 1;
```

Comment:
when do we need the 1d case?

Reply:
when we pipeline barriers and things like that.